RFS/src/main/java/org/opensearch/migrations/bulkload/lucene/LuceneDocumentsReader9.java

```diff
@@ -0,0 +1,230 @@
+package org.opensearch.migrations.bulkload.lucene;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+import org.opensearch.migrations.bulkload.common.RfsLuceneDocument;
+import org.opensearch.migrations.bulkload.common.Uid;
+
+import lombok.Lombok;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.reactivestreams.Publisher;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.scheduler.Schedulers;
+import shadow.lucene9.org.apache.lucene.document.Document;
+import shadow.lucene9.org.apache.lucene.index.DirectoryReader;
+import shadow.lucene9.org.apache.lucene.index.IndexReader;
+import shadow.lucene9.org.apache.lucene.index.LeafReaderContext;
+import shadow.lucene9.org.apache.lucene.index.SoftDeletesDirectoryReaderWrapper;
+import shadow.lucene9.org.apache.lucene.store.FSDirectory;
+import shadow.lucene9.org.apache.lucene.util.BytesRef;
+
+@RequiredArgsConstructor
+@Slf4j
+public class LuceneDocumentsReader9 implements LuceneDocumentsReader {
```

Comment:
I missed in the previous review that we duplicated the majority of the behavior from the other reader so now we have two copies.  How much was changed in this version vs. the other?  If it's not too much work, maybe we can centralize the core behavior in this PR; seems like that should be a light lift.

Reply:
I've created a task https://opensearch.atlassian.net/browse/MIGRATIONS-2373 for following up, it should be a prereq for lucene6 support.